pyomo/contrib/doe/doe.py

```diff
@@ -1521,17 +1525,22 @@ def compute_FIM_full_factorial(
             # Compute and record metrics on FIM
             D_opt = np.log10(np.linalg.det(FIM))
             A_opt = np.log10(np.trace(FIM))
-            E_vals, E_vecs = np.linalg.eig(FIM)  # Grab eigenvalues
-            E_ind = np.argmin(E_vals.real)  # Grab index of minima to check imaginary
+            E_vals =np.linalg.eigvalsh(FIM) # np.linalg.eigvalsh(FIM)  # Grab eigenvalues
+
+            #Shuvo: ????????????????????????
+            # Probably not required?
+            E_ind = 0 # np.argmin(E_vals.real)  # Grab index of minima to check imaginary
             # Warn the user if there is a ``large`` imaginary component (should not be)
             if abs(E_vals.imag[E_ind]) > 1e-8:
                 self.logger.warning(
-                    "Eigenvalue has imaginary component greater than 1e-6, contact developers if this issue persists."
+                    "Eigenvalue has imaginary component greater than 1e-8, contact developers if this issue persists."
```

Comment:
nit: this is why "embedded constants are evil".  An improved solution would be to declare a module-level constant (maybe `IMAG_THRESHOLD`) and then use that both in the test and in the message.